src/utl/src/timer.cpp

```diff
@@ -74,4 +76,23 @@ DebugScopedTimer::~DebugScopedTimer()
   debugPrint(logger_, tool_, group_.c_str(), level_, msg_, *this);
 }
 
+ScopedStatistics::ScopedStatistics(utl::Logger* logger)
+    : Timer(), logger_(logger)
+{
+}
+
+size_t ScopedStatistics::getPeakMemoryUsage()
+{
+  struct rusage rusage;
+  getrusage(RUSAGE_SELF, &rusage);
+  return (size_t) (rusage.ru_maxrss * 1024L);
+}
+
+ScopedStatistics::~ScopedStatistics()
+{
+  logger_->report("Runtime {} seconds, memory used {} KB.",
```

Comment:
This output leaves me with questions...

Scope? What took this long and used this much memory. What was the memory usage at start of scope compared to the peak usage.

Reply:
This doesn't describe the scope, what was run during this time.

Reply:
DRT already outputs in this format
```
[INFO DRT-0267] cpu time = 00:00:04, elapsed time = 00:00:01, memory = 2197.27 (MB), peak = 4031.35 (MB)
```
Is there a reason to have another?

Reply:
That prefix is nice and gives some context, but there should be a human description of the scope of what is being measured. In some cases it would be unmabigious from that prefix, but e.g. detailed routing measures for each iteration, for instance. So some further description of context could be useful.

Reply:
We can also add the command name before reporting its runtime to be clear about the scope.

Reply:
I'm fine to add a prefix

Reply:
The command name before reporting its runtime is already added. But by adding prefix like this `[INFO DRT-0267]` we should also create a unique message ID for each tool as `Logger::info` requires it, is this your suggestion?